src/components/WorkspaceDropdown/WorkspaceDropdown.jsx

```diff
@@ -0,0 +1,148 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import { useSnackbar } from '../Snackbar';
+import Dropdown, { DropdownSection, DropdownItem } from '../Dropdown';
+import Avatar from '../Avatar';
+
+import LoadingWorkspace from './Loading';
+
+import {
+  Trigger,
+  WorkspaceName,
+  Icon,
+  User,
+  UserInfo,
+  FullName,
+  Email,
+  BlankButton
+} from './Layout';
+
+export function PureWorkspaceDropdown({
+  isLoading,
+  refs,
+  workspaceName,
+  userAvatarUrl,
+  userFullName,
+  userEmail,
+  logout
+}) {
+  if (isLoading) {
+    return <LoadingWorkspace />;
+  }
+
+  return (
+    <Dropdown
+      width="220px"
+      refs={refs}
+      triggerEl={
+        <Trigger>
+          <WorkspaceName title={workspaceName}>{workspaceName}</WorkspaceName>
+          <Icon icon="chevron-down" size="sm" />
+        </Trigger>
+      }
+    >
+      <DropdownSection>
+        <User>
+          <Avatar
+            size="40px"
+            title={userFullName}
+            fullName={userFullName}
+            avatarUrl={userAvatarUrl}
+          />
+
+          <UserInfo>
+            <FullName title={userFullName}>{userFullName}</FullName>
+            <Email title={userEmail}>{userEmail}</Email>
+          </UserInfo>
+        </User>
+      </DropdownSection>
+
+      <DropdownSection>
+        <DropdownItem>
+          <BlankButton ref={refs[0]} role="menuitem" onClick={logout}>
+            Logout
+          </BlankButton>
+        </DropdownItem>
+      </DropdownSection>
+    </Dropdown>
+  );
+}
+
+PureWorkspaceDropdown.defaultProps = {
+  workspaceName: 'Workspace',
+  userFullName: 'User',
+  userEmail: 'email'
+};
+
+PureWorkspaceDropdown.propTypes = {
+  isLoading: PropTypes.bool.isRequired,
+  refs: PropTypes.array.isRequired,
+  workspaceName: PropTypes.string,
+  userAvatarUrl: PropTypes.string,
+  userFullName: PropTypes.string,
+  userEmail: PropTypes.string,
+  logout: PropTypes.func.isRequired
+};
+
+function WorkspaceDropdown({ user, workspace, logout }) {
+  const [, snackbarDispatch] = useSnackbar();
+
+  const { fetchErr } = workspace;
+  React.useEffect(() => {
+    if (!fetchErr) {
+      return;
+    }
+
+    snackbarDispatch({
+      type: 'ENQUEUE_SNACKBAR_MSG',
+      data: {
+        type: 'error',
+        title: 'Failed to fetch workspace',
+        text: fetchErr.details
+          ? `${fetchErr.message}: ${fetchErr.details}`
+          : fetchErr.message
+      }
+    });
+  }, [fetchErr]); // eslint-disable-line react-hooks/exhaustive-deps
+
+  const dropdownRefs = [React.createRef()];
```

Comment:
Move refs to pure component; the amount of refs depends on the amount of "focusable" buttons, which we only "know" from "inside" the pure component.